proto/trip.proto

```diff
@@ -232,6 +232,11 @@ message TripLeg {
     optional bool destination_only = 46;
     optional bool is_urban = 47; // uses edge density to decide if edge is in an urban area
     repeated TaggedName tagged_name = 48;
+
+    // for the part of the edge that is used in the path we must know where
+    // it starts and ends along the length of the edge as a percentage
+    optional float source_along_edge = 49;
+    optional float target_along_edge = 50;
```

Comment:
this is foreshadowing... there is currently a problem with recosting a path in that it only cares about partial edges at the beginning and end of a leg. sadly this isnt always the case. specifically  you can do uturns in the middle of an edge in the middle of a leg or you can have discontinuities on edges in the middle of a leg for map matching. in this case the recosting will not be correct for such edges (it will use the full cost of the edge). i'll make a separate issue for this.